src/sentry/api/endpoints/organization_release_details.py

```diff
@@ -12,20 +17,274 @@
     ReleaseHeadCommitSerializerDeprecated,
     ReleaseSerializer,
 )
-from sentry.models import Activity, Project, Release, ReleaseCommitError
+from sentry.models import Activity, Project, Release, ReleaseCommitError, ReleaseStatus
 from sentry.models.release import UnsafeReleaseDeletion
-from sentry.snuba.sessions import STATS_PERIODS, get_release_sessions_time_bounds
+from sentry.snuba.sessions import (
+    STATS_PERIODS,
+    get_adjacent_releases_based_on_adoption,
+    get_release_sessions_time_bounds,
+)
 from sentry.utils.sdk import bind_organization_context, configure_scope
 
 
+class InvalidSortException(Exception):
+    pass
+
+
 class OrganizationReleaseSerializer(ReleaseSerializer):
     headCommits = ListField(
         child=ReleaseHeadCommitSerializerDeprecated(), required=False, allow_null=False
     )
     refs = ListField(child=ReleaseHeadCommitSerializer(), required=False, allow_null=False)
 
 
-class OrganizationReleaseDetailsEndpoint(OrganizationReleasesBaseEndpoint, ReleaseAnalyticsMixin):
+def add_query_filter_to_queryset(queryset, query):
+    """
+    Function that adds a query filtering to a queryset
+    """
+    if query:
+        query_q = Q(version__icontains=query)
+
+        suffix_match = _release_suffix.match(query)
+        if suffix_match is not None:
+            query_q |= Q(version__icontains="%s+%s" % suffix_match.groups())
```

Comment:
This is potentially fairly slow on customers with many releases.

Reply:
separated into its own PR.. mentioned below!